src/rpcblockchain.cpp

```diff
@@ -1153,25 +1153,25 @@ bool CPIDAcidTest2(std::string bpk, std::string externalcpid)
 }
         
 
-bool AdvertiseBeacon(bool bFromService, std::string &sOutPrivKey, std::string &sOutPubKey, std::string &sError, std::string &sMessage)
+bool AdvertiseBeacon(std::string &sOutPrivKey, std::string &sOutPubKey, std::string &sError, std::string &sMessage)
 {	
      LOCK(cs_main);
      {
             GetNextProject(false);
             if (GlobalCPUMiningCPID.cpid=="INVESTOR")
             {
                 sError = "INVESTORS_CANNOT_SEND_BEACONS";
-                return bFromService ? true : false;
+                return false;
             }
 
             //If beacon is already in the chain, exit early
-            std::string sBeaconPublicKey = GetBeaconPublicKey(GlobalCPUMiningCPID.cpid,bFromService);
+            std::string sBeaconPublicKey = GetBeaconPublicKey(GlobalCPUMiningCPID.cpid,true);
             if (!sBeaconPublicKey.empty()) 
             {
                 // Ensure they can re-send the beacon if > 5 months old : GetBeaconPublicKey returns an empty string when > 5 months: OK.
                 // Note that we allow the client to re-advertise the beacon in 5 months, so that they have a seamless and uninterrupted keypair in use (prevents a hacker from hijacking a keypair that is in use)		
                 sError = "ALREADY_IN_CHAIN";
-                return bFromService ? true : false;
```

Comment:
This does change one thing though. Where we previously set the `bFromService` to true this would return true, but now it returns false. This causes the call in `GridcoinServices` to fail and an error will be shown. Come to think of it, that's actually an improvement.